praw/models/reddit/subreddit.py

```diff
@@ -2111,6 +2111,115 @@ def inbox(self, **generator_kwargs):
+        :param key_color: A 6-digit rgb hex color (e.g. ``"#AABBCC"``), used as a
+            thematic color for your subreddit on mobile.
+        :param lang: A valid IETF language tag (underscore separated).
+        :param link_type: The types of submissions users can make. One of ``any``,
+            ``link``, ``self``.
+        :param original_content_tag_enabled: Enables the use of the ``original content``
+            label for submissions.
+        :param over_18: Viewers must be over 18 years old (i.e. NSFW).
+        :param public_description: Public description blurb. Appears in search results
+            and on the landing page for private subreddits.
+        :param public_traffic: Make the traffic stats page public.
+        :param restrict_commenting: Specifies whether approved users have the ability to
+            comment.
+        :param restrict_posting: Specifies whether approved users have the ability to
+            submit posts.
+        :param show_media: Show thumbnails on submissions.
+        :param show_media_preview: Expand media previews on comments pages.
+        :param spam_comments: Spam filter strength for comments. One of ``all``,
+            ``low``, ``high``.
+        :param spam_links: Spam filter strength for links. One of ``all``, ``low``,
+            ``high``.
+        :param spam_selfposts: Spam filter strength for selfposts. One of ``all``, `
+            `low``, ``high``.
+        :param spoilers_enabled: Enable marking posts as containing spoilers.
+        :param submit_link_label: Custom label for submit link button
+            (None for default).
+        :param submit_text: Text to show on submission page.
+        :param submit_text_label: Custom label for submit text post button (None for
+            default).
+        :param subreddit_type: One of ``archived``, ``employees_only``, ``gold_only``,
+            ``gold_restricted``, ``private``, ``public``, ``restricted``.
+        :param suggested_comment_sort: All comment threads will use this sorting method
+            by default. Leave None, or choose one of ``confidence``, ``controversial``,
+            ``live``, ``new``, ``old``, ``qa``, ``random``, ``top``.
+        :param title: The title of the subreddit.
+        :param welcome_message_enabled: Enables the subreddit welcome message.
+        :param welcome_message_text: The text to be used as a welcome message. A welcome
+            message is sent to all new subscribers by a Reddit bot.
+        :param wiki_edit_age: Account age, in days, required to edit and create wiki
+            pages.
+        :param wiki_edit_karma: Subreddit karma required to edit and create wiki pages.
+        :param wikimode: One of  ``anyone``, ``disabled``, ``modonly``.
+
+        Additional keyword arguments can be provided to handle new settings as
+        Reddit introduces them.
+
+        Settings that are documented here and aren't explicitly set by you in a
```

Comment:
Should we add a see also for `updated` and list a few of the values which are not updatable by this method? And similarly update the documentation for `update` with a see also to this method and point out which fields can only be updated by `legatcy_update`?

Reply:
Yes. I can compile a list of settings that don't work for each.

Reply:
I did some further analysis on this.

I ran the following code:
```python
subreddit = reddit.subreddit("Lil_SpazJoekp")
failed = {}
success = {}

current_settings = subreddit.mod.settings()

setting_remap = {
    "content_options": "link_type",
    "default_set": "allow_top",
    "header_hover_text": "header_title",
    "language": "lang",
    "subreddit_type": "type",
}

for key, value in sorted(current_settings.items(), key=lambda d: d[0]):
    key = setting_remap.get(key, key)
    if key == 'subreddit_id': # subreddit_id is not changeable lol
        print("Skipped: subreddit_id")
        continue
    result = subreddit.mod.update(**{key: value})
    if result:
        success[key] = value
        print(f"Passed: {key}, response: {result}")
    else:
        failed[key] = value
        print(f"Failed: {key}, response: {result}")
print(f'\nFailed settings: {failed}')
```

<details>
<summary>The subreddit settings to start off with</summary>
<p>

```python
{
    'all_original_content': False,
    'allow_chat_post_creation': True,
    'allow_discovery': False,
    'allow_galleries': True,
    'allow_images': True,
    'allow_polls': True,
    'allow_post_crossposts': True,
    'allow_predictions': False,
    'allow_predictions_tournament': False,
    'allow_videos': True,
    'collapse_deleted_comments': False,
    'comment_score_hide_mins': 0,
    'content_options': 'any',
    'crowd_control_chat_level': 1,
    'crowd_control_level': 0,
    'crowd_control_mode': False,
    'default_set': True,
    'description': '',
    'disable_contributor_requests': False,
    'domain': None,
    'exclude_banned_modqueue': False,
    'free_form_reports': True,
    'header_hover_text': 'header_hover_text',
    'hide_ads': False,
    'key_color': '#545452',
    'language': 'en',
    'new_pinned_post_pns_enabled': True,
    'original_content_tag_enabled': True,
    'over_18': False,
    'prediction_leaderboard_entry_type': 1,
    'public_description': 'Lil_SpazJoekp’s Personal Subreddit',
    'public_traffic': False,
    'restrict_commenting': False,
    'restrict_posting': True,
    'show_media': True,
    'show_media_preview': True,
    'spam_comments': 'low',
    'spam_links': 'high',
    'spam_selfposts': 'high',
    'spoilers_enabled': True,
    'submit_link_label': '',
    'submit_text': '',
    'submit_text_label': '',
    'subreddit_id': 't5_7ss0q',
    'subreddit_type': 'restricted',
    'suggested_comment_sort': None,
    'title': 'Lil_SpazJoekp',
    'toxicity_threshold_chat_level': 1,
    'user_flair_pns_enabled': True,
    'welcome_message_enabled': False,
    'welcome_message_text': '',
    'wiki_edit_age': 0,
    'wiki_edit_karma': 100,
    'wikimode': 'modonly'
}
```
</p>
</details>

This is the output of the above snippet:
```
Passed: all_original_content, response: {'all_original_content': False}
Passed: allow_chat_post_creation, response: {'allow_chat_post_creation': True}
Passed: allow_discovery, response: {'allow_discovery': False}
Passed: allow_galleries, response: {'allow_galleries': True}
Passed: allow_images, response: {'allow_images': True}
Passed: allow_polls, response: {'allow_polls': True}
Passed: allow_post_crossposts, response: {'allow_post_crossposts': True}
Passed: allow_predictions, response: {'allow_predictions': False}
Passed: allow_predictions_tournament, response: {'allow_predictions_tournament': False}
Passed: allow_videos, response: {'allow_videos': True}
Passed: collapse_deleted_comments, response: {'collapse_deleted_comments': False}
Passed: comment_score_hide_mins, response: {'comment_score_hide_mins': 0}
Passed: link_type, response: {'link_type': 'any'}
Passed: crowd_control_chat_level, response: {'crowd_control_chat_level': 1}
Passed: crowd_control_level, response: {'crowd_control_level': None}
Passed: crowd_control_mode, response: {'crowd_control_mode': False}
Passed: allow_top, response: {'allow_top': True}
Failed: description, response: {}
Passed: disable_contributor_requests, response: {'disable_contributor_requests': False}
Failed: domain, response: {}
Passed: exclude_banned_modqueue, response: {'exclude_banned_modqueue': False}
Passed: free_form_reports, response: {'free_form_reports': True}
Passed: header_title, response: {'header_title': 'header_hover_text'}
Passed: hide_ads, response: {'hide_ads': False}
Passed: key_color, response: {'key_color': '#545452'}
Passed: lang, response: {'lang': 'en'}
Passed: new_pinned_post_pns_enabled, response: {'new_pinned_post_pns_enabled': True}
Passed: original_content_tag_enabled, response: {'original_content_tag_enabled': True}
Passed: over_18, response: {'over_18': False}
Passed: prediction_leaderboard_entry_type, response: {'prediction_leaderboard_entry_type': 1}
Passed: public_description, response: {'public_description': 'Lil_SpazJoekp’s Personal Subreddit'}
Failed: public_traffic, response: {}
Passed: restrict_commenting, response: {'restrict_commenting': False}
Passed: restrict_posting, response: {'restrict_posting': True}
Passed: show_media, response: {'show_media': True}
Passed: show_media_preview, response: {'show_media_preview': True}
Passed: spam_comments, response: {'spam_comments': 'low'}
Passed: spam_links, response: {'spam_links': 'high'}
Passed: spam_selfposts, response: {'spam_selfposts': 'high'}
Passed: spoilers_enabled, response: {'spoilers_enabled': True}
Passed: submit_link_label, response: {'submit_link_label': None}
Passed: submit_text, response: {'submit_text': ''}
Passed: submit_text_label, response: {'submit_text_label': None}
Skipped: subreddit_id
Passed: type, response: {'type': 'restricted'}
Passed: suggested_comment_sort, response: {'suggested_comment_sort': None}
Passed: title, response: {'title': 'Lil_SpazJoekp'}
Passed: toxicity_threshold_chat_level, response: {'toxicity_threshold_chat_level': 1}
Passed: user_flair_pns_enabled, response: {'user_flair_pns_enabled': True}
Passed: welcome_message_enabled, response: {'welcome_message_enabled': False}
Passed: welcome_message_text, response: {'welcome_message_text': ''}
Passed: wiki_edit_age, response: {'wiki_edit_age': 0}
Passed: wiki_edit_karma, response: {'wiki_edit_karma': 100}
Passed: wikimode, response: {'wikimode': 'modonly'}

Failed settings: {'description': '', 'domain': None, 'public_traffic': False}
```

For the failed settings are:
- `description` this is the sidebar on old reddit and appears to be the only setting that isn't editable.
- `domain` not sure what this is, but it is set to `None`.
- `public_traffic` this one is self-explanatory however, it is not able to be edited on all the subreddits I've tested with.

With what I've found above, I have a couple of ideas:
1. Add a method specifically for updating the sidebar.
2. Modify the existing `update()` method to hit a different endpoint if `description` is being changed.
3. Add the `legacy_update()` method for updating subreddit settings using the documented endpoint.

Personally, I feel that the second option would be the best since all the settings, except for `description`, can be edited using the current `update()` method.

Reply:
Copying this conversation for posterity:

![image](https://user-images.githubusercontent.com/48100/108548886-45b93980-72a1-11eb-8a37-2a8a7ec07759.png)